src/scripts/main.js

```diff
@@ -1,11 +1,51 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener('click', (e) => {
+      // our event is click
+      if (e.target === element) {
+        // when our elemnt is
+        resolve(
+          // element wich called event we resolve promise
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        ); // we resolve string with data
+      }
+    });
+
+    element.addEventListener('contextmenu', (e) => {
+      // we exchange our event
+      if (e.target === element) {
+        // we doing the same
+        resolve(
+          // check out if element is that what we need
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    });
+
+    element.addEventListener('blur', (e) => {
+      // also use it with blur
+      if (e.target === element) {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    });
```

Comment:
The `waitFor` function currently listens for 'click', 'contextmenu', and 'blur' events regardless of the `eventName` passed as an argument. This is not dynamic and does not utilize the `eventName` parameter effectively. Consider using `element.addEventListener(eventName, ...)` to dynamically listen for the event specified by `eventName`.